### PR TITLE
Promisify

### PR DESCRIPTION
**Pull request by SamVerschueren**
I promisified the library and converted the tests to use ES2015.

I don't have a linux computer so I couldn't tested the script.

### CODE REVIEW

**Comment by kevva** [comment]
Thanks, I'll check it out :).

File: readme.md [hunk 1]
```diff
@@ -15,12 +15,12 @@ $ npm install --save xdg-trashdir
...
+xdgTrashdir().then(function (dir) {
 	console.log(dir);
 	//=> /home/johndoe/.local/share/Trash
 });
 
-xdgTrashdir('foo.zip', function (err, dir) {
+xdgTrashdir('foo.zip').then(function (dir) {
 	console.log(dir);
 	//=> /media/johndoe/UUI/.Trash-1000
 });
```
Comment (SamVerschueren) [inline readme.md]: Sindre wants it to be this way:

```js
const xdgTrashdir = require('xdg-trashdir');

xdgTrashdir().then(dir => {
	console.log(dir);
	//=> /home/johndoe/.local/share/Trash
});

xdgTrashdir('foo.zip').then(dir => {
	console.log(dir);
	//=> /media/johndoe/UUI/.Trash-1000
});
```

Let me know what you prefer :)

File: test/test.js [hunk 2]
```diff
@@ -1,64 +1,46 @@
...
-test('get the trash path', function (t) {
-	t.plan(2);
+const exec = pify(execFile);
 
-	trashdir(function (err, dir) {
-		t.assert(!err, err);
-		t.assert(dir === path.join(xdgBasedir.data, 'Trash'), dir);
-	});
+test('get the trash path', async t => {
+	var dir = await trashdir();
```
Comment (kevva) [inline test/test.js]: `var` => `const`

File: readme.md [hunk 3]
```diff
@@ -15,12 +15,12 @@ $ npm install --save xdg-trashdir
...
 var xdgTrashdir = require('xdg-trashdir');
 
-xdgTrashdir(function (err, dir) {
+xdgTrashdir().then(function (dir) {
 	console.log(dir);
 	//=> /home/johndoe/.local/share/Trash
 });
 
-xdgTrashdir('foo.zip', function (err, dir) {
+xdgTrashdir('foo.zip').then(function (dir) {
```
Comment (kevva) [inline readme.md]: `xdgTrashdir('foo.zip').then(dir => {`

File: readme.md [hunk 4]
```diff
@@ -15,12 +15,12 @@ $ npm install --save xdg-trashdir
 ```js
 var xdgTrashdir = require('xdg-trashdir');
 
-xdgTrashdir(function (err, dir) {
+xdgTrashdir().then(function (dir) {
```
Comment (kevva) [inline readme.md]: `xdgTrashdir('foo.zip').then(dir => {`

File: readme.md (hunk 1 above)
Comment (kevva) [inline readme.md]: Oh, I didn't see this comment, but yeah, that's what I prefer too :).

File: index.js [hunk 5]
```diff
@@ -4,61 +4,47 @@ var path = require('path');
...
-				cb(err);
-				return;
-			}
-
-			if (res.mount === ret.mount) {
-				cb(null, path.join(xdgBasedir.data, 'Trash'));
-				return;
-			}
+	return Promise.all([
+		pify(mountPoint, Promise)(userHome),
```
Comment (sindresorhus) [inline index.js]: pify it once at the top of the file and reuse it here

File: test/test.js [hunk 6]
```diff
@@ -1,64 +1,46 @@
-'use strict';
-var execFile = require('child_process').execFile;
-var path = require('path');
-var test = require('ava');
-var xdgBasedir = require('xdg-basedir');
-var trashdir = require('../');
+import {execFile} from 'child_process';
+import path from 'path';
+import test from 'ava';
+import xdgBaseDir from 'xdg-basedir';
```
Comment (kevva) [inline test/test.js]: `xdgBaseDir` => `xdgBasedir`

Comment (SamVerschueren) [inline test/test.js]: Good catch :)

**Comment by SamVerschueren** [comment]
Should we remove Node 0.10 tests out of travis?

**Comment by kevva** [comment]
@SamVerschueren, yup, I did it in the same commit. Thank you :D!

**Comment by SamVerschueren** [comment]
Nice! Thanks for merging!

**Pull request closed without merging**